applications/vanilla/views/vanillasettings/posting.php

```diff
@@ -23,6 +29,52 @@
             echo $form->toggle('Garden.Format.DisableUrlEmbeds', $embedsLabel, [], $embedsDesc, true);
             ?>
         </li>
+
+        <li class="form-group">
+            <?php
+            $formatNotes1 = t('InputFormatter.Notes1', 'Select the default format of the editor for posts in the community.');
+            $formatNotes2 = t('InputFormatter.Notes2', 'The editor will auto-detect the format of old posts when editing them and load their 
+            original formatting rules. Aside from this exception, the selected post format below will take precedence.');
+            $label = '<p class="info">'.$formatNotes1.'</p><p class="info"><strong>'.t('Note:').' </strong>'.$formatNotes2.'</p>';
+            ?>
+            <div class="label-wrap">
+                <?php
+                echo $form->label('Post Format', 'Garden.InputFormatter');
+                echo $label;
+                ?>
+            </div>
+            <div class="input-wrap">
+                <?php echo $form->dropDown('Garden.InputFormatter', $formatOptions); ?>
+            </div>
+        </li>
+        <li class="form-group">
+            <?php
+            $forceWysiwygLabel = 'Reinterpret All Posts As Wysiwyg';
+            $forceWysiwygNote1 =  t('ForceWysiwyg.Notes1', 'Check the below option to tell the editor to reinterpret all old posts as Wysiwyg.');
+            $forceWysiwygNote2 = t('ForceWysiwyg.Notes2', 'This setting will only take effect if Wysiwyg was chosen as the Post Format above. The purpose of this option is to normalize the editor format. If older posts edited with another format, such as markdown or BBCode, are loaded, this option will force Wysiwyg.');
+            $label = '<p class="info">'.$forceWysiwygNote1.'</p><p class="info"><strong>'.t('Note:').' </strong>'.$forceWysiwygNote2.'</p>';
+            echo $form->toggle('Plugins.editor.ForceWysiwyg', $forceWysiwygLabel, [], $label);
```

Comment:
Does this addon-specific config value need to be in a core settings page? Do you think any other addons might use this value? If so, it might be worth creating a core counterpart. If not, and it's specific to Advanced Editor, maybe we could move this setting back into Advanced Editor's settings page.

Reply:
It is directly tied to and appears on this page when the wysiwyg format is enabled (handled in our existing dashboard JS). Making this the only Advanced Editor settings (likely ever again) doesn't seem right. I figure if/when we offer an equivalent conversion method for the Rich Editor we can unify it under 1 global setting. Likely `Garden.ConvertLegacyFormattingOptions` or something like that.

 I didn't want to change this value because it is likely being used somewhere that it shouldn't be and I didn't want to blow up the scope by tracking down all usages of this config value and ensuring that existing values carry over.

Reply:
If you want to keep it on this page, that's cool, but I'm still not onboard with hardcoding addon configs into core settings pages. This setting is seemingly dependent on the Advanced Editor being enabled. I think there's the potential for real confusion if someone doesn't have the Advanced Editor enabled, but is trying to make use of this config. What about hooking in from Advanced Editor to add the field to this form?